py-scripts/lf_ftp.py

```diff
@@ -2096,7 +2484,58 @@ def pass_fail_duration(band, file_size):
             duration = float(duration)
 
         return duration
-
+    # Get group and profile values from arguments and convert comma-separated strings into lists
+    if (args.group_name is not None):
+        selected_groups = args.group_name.split(',')
+    else:
+        selected_groups = []  # Default to empty list if group name is not provided
+    if (args.profile_name is not None):
+        selected_profiles = args.profile_name.split(',')
+    else:
+        selected_profiles = []  # Default to empty list if profile name is not provided
+
+    if args.clients_type == 'Real' and args.config and args.group_name is None:
+        if args.ssid and args.security and args.security.lower() == 'open' and (args.passwd is None or args.passwd == ''):
+            args.passwd = '[BLANK]'
+        if args.ssid is None:
+            logger.info('Specify SSID for confiuration, Password(Optional for "open" type security) , Security')
+            exit(0)
+        elif args.ssid and args.passwd:
+            if args.security is None:
+                logger.info('Security must be provided when SSID and Password specified')
+                exit(0)
+            elif args.security.lower() == 'open' and args.passwd != '[BLANK]':
+                logger.info("For a open type security there will be no password or the password should be left blank (i.e., set to '' or [BLANK]).")
+                exit(0)
+        elif args.ssid and args.passwd == '[BLANK]' and args.security and args.security.lower() != 'open':
+            logger.info('Please provide valid passwd and security configuration')
+            exit(0)
+
+    if (len(selected_groups) != len(selected_profiles)):
+        logger.warning("Number of groups should match number of profiles")
+        exit(0)
+    elif ((args.group_name is not None and (args.file_name is None or args.profile_name is None)) or (args.file_name is not None and (args.group_name is None or args.profile_name is None)) or (args.profile_name is not None and (args.group_name is None or args.file_name is None))):
```

Comment:
See my earlier feedback on checking for not None. Because these checks use the more verbose form of checks, which is discouraged, this conditional is very long and difficult to read.